finder-app/finder.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if [ $# -lt 2 ]
+then
+	echo filesdir and searchstr were not specified, exitting.
+	echo Usage: ./finder.sh /path/to/directory/to/search string_to_search.
+	exit 1
+fi
+
+filesdir=$1
+searchstr=$2
+
+if [ -d $filesdir ]
+then 
+	cd $filesdir
+else
+	echo $filesdir is not a valid directory on the file system, exitting.
+	echo Please enter a valid directory/path.
+	exit 1
+fi
+
+i=0
+j=0
+
+matchlist=$(grep -r $searchstr -c | cut -d":" -f2)
```

Comment:
@DhruvHMehta I'm curoius why the logic with cut here?

Reply:
I noticed in the terminal that my grep command returned a list of all files in each directory with the colon as the delimiter for the file count. So I used cut to get the second column and add it up in the for loop. In hindsight, I should have read the man page for grep, I believe there was an easier way.
Eg. file.txt:2 , cut gets the 2 and then the for loop adds them all up to get the final match count. Hope this answers your question @dwalkes .